src/OpenTelemetry/Trace/ActivityProcessor.cs

```diff
@@ -49,5 +52,24 @@ public abstract class ActivityProcessor
         /// <param name="cancellationToken">Cancellation token.</param>
         /// <returns>Returns <see cref="Task"/>.</returns>
         public abstract Task ForceFlushAsync(CancellationToken cancellationToken);
+
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            this.Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            try
```

Comment:
I guess this block should only be called when `disposing == true`?

Reply:
in our actual code, it's doing every time. So, I'm not sure.

Reply:
I think typically Close/Shutdown call goes outside of the check. Eg:
```csharp
protected virtual void Dispose(bool disposing)
{
   Close();

   if (disposing)
   {
       // Free managed resources.
   }
}
```

Reply:
after I moved, i saw some problems from BroadcastActivityProcessor:
```csharp
protected override void Dispose(bool disposing)
        {
            if (disposing && !this.disposed)
            {
                foreach (var processor in this.processors)
                {
                    try
                    {
                        if (processor is IDisposable disposable)
                        {
                            disposable.Dispose();
                        }
                    }
                    catch (Exception e)
                    {
                        OpenTelemetrySdkEventSource.Log.SpanProcessorException("Dispose", e);
                    }
                }

                this.disposed = true;
            }

            base.Dispose(disposing);
        }
```

we are disposing the processors and, then, the base.dispose will call the shutdown...

I will revert to the base calling before, and we can think in another way.